src/Microsoft.Graph.Core/Requests/BaseClient.cs

```diff
@@ -17,7 +17,7 @@ namespace Microsoft.Graph
     /// <summary>
     /// A default client implementation.
     /// </summary>
-    public class BaseClient
```

Comment:
I think this class should be moved to the service lib, renamed GraphServiceClient and inherit from the GraphBaseServiceClient (generated by Kiota) for the full loop to be closed on this. Otherwise you'll end up needing to inherit from 2 classes in the service lib, not possible in dotnet.

Reply:
Actually, this class is only meant to be used internally in the core lib really. For testing and when a concrete implementation of the interface is needed. 
I've made it internal to stay consistent with the plan if that makes sense.

Reply:
thanks for the clarification!